packages/block-library/src/image/interactivity.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Internal dependencies
+ */
+import { store } from '../utils/interactivity';
+
+const raf = window.requestAnimationFrame;
+// Until useSignalEffects is fixed: https://github.com/preactjs/signals/issues/228
+const tick = () => new Promise( ( r ) => raf( () => raf( r ) ) );
+
+store( {
+	actions: {
+		core: {
+			image: {
+				showLightbox: ( { context } ) => {
+					context.core.image.initialized = true;
+					context.core.image.lightboxEnabled = true;
+					context.core.image.lastFocusedElement =
+						window.document.activeElement;
+					context.core.image.scrollPosition = window.scrollY;
+				},
+				hideLightbox: ( { context, event } ) => {
+					if ( context.core.image.lightboxEnabled ) {
+						// If scrolling, wait a moment before closing the lightbox.
+						if (
+							event.type === 'mousewheel' &&
+							Math.abs(
+								window.scrollY -
+									context.core.image.scrollPosition
+							) < 5
+						) {
+							return;
+						}
+						context.core.image.lightboxEnabled = false;
+
+						// We only want to focus the last focused element
+						// if the lightbox was closed by the keyboard.
+						// Note: Pressing Enter on a button will trigger
+						// a click event with a blank pointerType.
+						if (
+							( event.key && event.type === 'keydown' ) ||
+							( event.type === 'click' &&
+								event.pointerType === '' )
+						) {
+							context.core.image.lastFocusedElement.focus();
+						}
+					}
+				},
+				handleKeydown: ( { context, actions, event } ) => {
+					if ( context.core.image.lightboxEnabled ) {
+						const isTabKeyPressed =
+							event.key === 'Tab' || event.keyCode === 9;
+						const escapeKeyPressed =
+							event.key === 'Escape' || event.keyCode === 27;
+
+						if ( isTabKeyPressed ) {
+							event.preventDefault();
+						}
+
+						if ( escapeKeyPressed || isTabKeyPressed ) {
```

Comment:
What's the rationale behind using a Tab key to close the lightbox?

I assumed that you the Escape key would trigger that, or pressing Enter or Space on the close button.

Reply:
+1 to @gziolo's comment. Only Esc, Enter, and Space should close the dialog.